src/command-line/install.js

```diff
@@ -53,59 +52,20 @@ program
 				}, null, "\t"));
 			}
 
-			const yarn = path.join(
-				__dirname,
-				"..",
-				"..",
-				"node_modules",
-				"yarn",
-				"bin",
-				"yarn.js"
-			);
-
-			let success = false;
-			const add = child.spawn(
-				process.execPath,
-				[
-					yarn,
-					"add",
-					"--json",
-					"--exact",
-					"--production",
-					"--ignore-scripts",
-					"--non-interactive",
-					"--cwd",
-					packagesPath,
-					`${json.name}@${json.version}`,
-				]
-			);
-
-			add.stdout.on("data", (data) => {
-				data.toString().trim().split("\n").forEach((line) => {
-					line = JSON.parse(line);
-
-					if (line.type === "success") {
-						success = true;
-					}
-				});
-			});
-
-			add.stderr.on("data", (data) => {
-				log.error(data.toString());
-			});
-
-			add.on("error", (e) => {
-				log.error(`${e}`);
-				process.exit(1);
-			});
-
-			add.on("close", (code) => {
-				if (!success || code !== 0) {
-					log.error(`Failed to install ${colors.green(json.name + " v" + json.version)}. Exit code: ${code}`);
-					return;
-				}
-
+			return Utils.executeYarnCommand(
+				"add",
+				"--json",
+				"--exact",
+				"--production",
+				"--ignore-scripts",
+				"--non-interactive",
+				"--cwd",
+				packagesPath,
+				`${json.name}@${json.version}`
+			).then(() => {
```

Comment:
Why did you go for rest params here? I'd find it clearer to have an array: `Utils.executeYarnCommand(["add", "--json", /* ... */ ])`

Reply:
I need to prepend the executable.

Reply:
What you're doing would still work: [yarn, ...parameters]

Reply:
How does having an array change anything then?

Reply:
It just seems clearer to pass a single array argument than an arbitrary number of parameters imo.
As far as I can tell, I mostly see this in console.log and array.push (very rarely).
No big deal though